libs/datamodel/core/src/ast/model.rs

```diff
@@ -3,7 +3,7 @@ use super::*;
 /// An opaque identifier for a field in an AST model. Use the
 /// `model[field_id]` syntax to resolve the id to an `ast::Field`.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub(crate) struct FieldId(u32);
+pub(crate) struct FieldId(pub(crate) u32);
```

Comment:
I think you made it `pub(crate)` so it can be used in `type_definition.rs`. Since we do want to keep it opaque, wdyt about using `pub(super)` or turning it back to private, but in the parent module?